orttraining/orttraining/python/training/ortmodule/experimental/hierarchical_ortmodule/_hierarchical_ortmodule.py

```diff
@@ -15,6 +18,40 @@
 )
 
 
+class IteratedORTModule(torch.nn.Module):
```

Comment:
Do you think we could instead provide a caching mechanism inside ortmodule that caches the exported graphs/inference session (and other dependencies) on the input types hash, and then based on that executes the graph.
I think users might also benefit if ortmodule is able to support multiple types without needing to re-export the model.

We can do this later considering this might take longer, but just wanted to hear your thoughts on this.

Reply:
We can think about it. Hash to graph is similar to hash to ORTModule, because each time we still need to export new graph and build gradients. We can think about the easiest way to implement.